explainer.md

```diff
@@ -0,0 +1,182 @@
+
+const client = new Client(
+    'http://127.0.0.1',			// <API_URL>
+    '4000',				// <API_PORT>
+    'ws://127.0.0.1:4000'		// <WEBSOCKET_ENDPOINT>
+);
+
+client.init().then(() => {
+    console.log('Client initialised');
+});
+```
+
+The `Client` is divided into two modes of communication with the [testchain-backendgateway](https://github.com/makerdao/testchain-backendgateway), a REST api for retrieving chain and snapshot data, and a websocket api for performing chain functions.
+
+### RxJS usage
+
+Using [RxJS](https://rxjs-dev.firebaseapp.com/), each channel, (effectively each chain), listens for a set of events (which can be found at the bottom of the document). The object that allows this is called an [`Observable`](https://rxjs-dev.firebaseapp.com/api/index/class/Observable) which is a set of these events over a period of time.
+
+##### Event constants
+
+The event constants we use in the system can be found [here](https://github.com/makerdao/testchain-client/blob/89136da8472acfa1e62c6e448cf6cff117f897a4/src/core/ChainEvent.js). These are to be used in conjunction with `once()` and `sequenceEvents()`
+
+##### `stream()`
+
+The Client's `stream()` function returns the `Observable` which we can then subscribe to and monitor the data stream coming from each channel.
+
+``` javascript
+const chainStream = client.stream(id); // Specify the chain by it's id
+const obs = chainStream.subscribe(
+    ({ event, payload }) => {
+        // do something
+    },
+    (err) => throw ...,	
+);
+.
+.
+.
+obs.unsubscribe();
+```
+In the above example, the RxJS Observable is assigned to our `chainStream` constant. `subscribe` takes two functions as arguments and are executed on an effective infinite loop for each incoming event fired from the backend on each channel.
+
+The first, is where all incoming data is passed and where a user should expect to find the chain events and returning data. The client has bootstrapped the `event` name to each returned payoad object to make it easier for the user to specify a target event. The second callback is the error event which is used in the event that a chain fires an error event (TODO).
+We can easily subscribe and unsubscribe from these events whenever we want.
+
+##### `once()`
+
+The `once()` function builds around this subscription model and wraps a promise based on the next incoming event, returning the payload. This is especially useful around events like `started` and `snapshot_taken` which return data based on their events.
+
+```javascript
+client.create({ ...args });
+const chain = await client.once('api', Event.CHAIN_STARTED);
+```
+The above awaits on a promise which resolves on the `Event.CHAIN_STARTED` event firing on the `api` channel with that payload data.
+
+`once()` can also accept a predicate function callback instead of an event string. This function takes both the incoming event and payload information on that channel and must return a boolean based on this information;
```

Comment:
It's important not call this a callback function, that's what originally got me confused, its a `predicate function`, not a function to be used to consume the event payload but a function to check if this is the event matching the predicate, this method is still a promise, so callbacks are handled using `then` or `await`

Reply:
Sorry about that, will change